extensions/http-curl/tests/C2DescribeManifestTest.cpp

```diff
@@ -41,8 +42,17 @@ class DescribeManifestHandler: public HeartbeatHandler {
 int main(int argc, char **argv) {
   const cmd_args args = parse_cmdline_args(argc, argv, "heartbeat");
   VerifyC2Describe harness;
+  utils::crypto::Bytes encryption_key = utils::StringUtils::from_hex("4024b327fdc987ce3eb43dd1f690b9987e4072e0020e3edf4349ce1ad91a4e38");
+  minifi::Decryptor decryptor{utils::crypto::EncryptionProvider{encryption_key}};
+  std::string encrypted_value = "l3WY1V27knTiPa6jVX0jrq4qjmKsySOu||ErntqZpHP1M+6OkA14p5sdnqJhuNHWHDVUU5EyMloTtSytKk9a5xNKo=";
+
+  harness.setConfiguration(std::make_shared<minifi::Configure>(decryptor));
   harness.setKeyDir(args.key_dir);
   DescribeManifestHandler responder(harness.getConfiguration());
+
+  harness.getConfiguration()->set(minifi::Configuration::nifi_rest_api_password, encrypted_value);
+  harness.getConfiguration()->set(std::string(minifi::Configuration::nifi_rest_api_password) + ".protected", utils::crypto::EncryptionType::name());
```

Comment:
HTTPHandlers.cpp:536 has also been updated I think that verifies that we receive the encrypted  value

Reply:
Yes, as @adamdebreceni mentionsed the verification in HTTPHandlers makes sure we check for the encrypted version in the json. I checked the test with both `get` and `getString` to make sure it fails if we are using the `get` function and succeeds if we expect the encrypted version with `getString`.

Reply:
I would have preferred something more explicit, as, for example, if someone changes `getString()` to return the decrypted value, the test will continue to pass.  On the other hand, it is unlikely someone would do this, and it would probably be caught by other tests, so I'm OK with this version, too.